staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go

```diff
@@ -263,7 +263,12 @@ func (o *ApplyOptions) Complete(f cmdutil.Factory, cmd *cobra.Command) error {
 	}
 
 	o.OpenAPISchema, _ = f.OpenAPISchema()
-	o.Validator, err = f.Validator(cmdutil.GetFlagBool(cmd, "validate"))
+
+	validate := cmdutil.GetFlagBool(cmd, "validate")
+	if o.ServerSideApply {
+		validate = false
```

Comment:
should we print an error instead? rather than ignore the flag?

Reply:
Hm we need to detect if `--validate` is true because it's default or if it's because someone set `--validate=true`, then we would error... I will try this, but I think I remember that cobra might not actually let us do this

Reply:
Ah yeah, that's not easy ...